### ISSUE TITLE
[Android] BUILD FAILED after using this command yarn add react-native-kit/react-native-track-player#dev

### ISSUE THREAD

**Issue by RavindraMaddukuri** [react-native-kit/react-native-track-player#626]
hi,
  when I am trying to build the app getting this error in android
```

> Task :react-native-track-player:compileReleaseJavaWithJavac FAILED
/Users/stellentsoft/Desktop/Musicapp/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java:10: error: package androidx.localbroadcastmanager.content does not exist
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
                                             ^
/Users/stellentsoft/Desktop/Musicapp/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java:45: error: cannot find symbol
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
        ^
  symbol:   variable LocalBroadcastManager
  location: class MusicService
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-track-player:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3s
24 actionable tasks: 10 executed, 14 up-to-date
StellentsMini3:android stellentsoft$ 

```

after updating the react native track player by using the command given below.

yarn add react-native-kit/react-native-track-player#dev

and my android/build.gradle file
`// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 16
        compileSdkVersion = 28
        targetSdkVersion = 28
        supportLibVersion = "28.0.0"
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath("com.android.tools.build:gradle:3.4.0")

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        google()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}`

**Comment by gazedash** [comment]
@RavindraMaddukuri Can you post your android/app/build.gradle as well as other files?
Did you do a react-native link? And did you follow instructions on how to install this library?

**Comment by RavindraMaddukuri** [comment]
hi, @gazedash I have followed all the linking instructions given in the documentation and here is my android/app/build.gradle
`
project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "com.musicapp"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation project(':react-native-gesture-handler')
    implementation project(':react-native-track-player')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation "com.facebook.react:react-native:+"  // From node_modules
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}
configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        def requested = details.requested
        if (requested.group == 'com.android.support') {
            if (!requested.name.startsWith("multidex")) {
                details.useVersion '26.0.1'
            }
        }
    }
}
`

**Comment by laercioc** [comment]
I have the same problem

`> Task :react-native-track-player:compileDebugJavaWithJavac FAILED
C:\Cold\node_modules\react-native-track-player\android\src\main\java\com\guichaguri\trackplayer\service\MusicService.java:10: error: package androidx.localbroadcastmanager.content does not exist
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
                                             ^
C:\Cold\node_modules\react-native-track-player\android\src\main\java\com\guichaguri\trackplayer\service\MusicService.java:45: error: cannot find symbol
        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
        ^
  symbol:   variable LocalBroadcastManager
  location: class MusicService
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors`

**Comment by roman-gula** [comment]
I have similar issue
<pre>
<b>> Task :react-native-track-player:compileDebugJavaWithJavac FAILED</b>
/.../node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java:9: error: cannot find symbol
import androidx.core.content.LocalBroadcastManager;
                            ^
  symbol:   class LocalBroadcastManager
  location: package androidx.core.content
/.../node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java:12: error: cannot find symbol
import androidx.core.content.LocalBroadcastManager;

...

Note: /.../node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/MusicManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
7 errors

FAILURE: Build failed with an exception.
</pre>

It occurred while I was trying to fix issues connected to [Android X migration](https://developers.google.com/android/guides/releases).

**Comment by roman-gula** [comment]
So, I found solution [here](https://github.com/airbnb/DeepLinkDispatch/issues/222#issuecomment-424427251).

I replaced 
`import androidx.core.content.LocalBroadcastManager;`
with
`import androidx.localbroadcastmanager.content.LocalBroadcastManager;`

in `MusicModule.java:9` and `MusicService.java:12` files and everything works fine.

**Comment by Guichaguri** [comment]
Fixed in 5960a37ddc67d5e33d4368b8cc6c7fa3a6183fba
